### ISSUE TITLE
after upgrade micronaut 2.4.1 -> 2.5.0 existsBy and findBy produce wrong where clause if another column with name id exist

### ISSUE THREAD

**Issue by StefanSchmidtOz** [micronaut-projects/micronaut-data#1008]
### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

In this example 

Micronaut 2.4.1 produces a query with `WHERE (someEntity_.transactid = :p1)`
Micronaut 2.5.0 produces a query with `WHERE (someEntity_.id = :p1)`

1. Create entity with 
```
package bug;
import io.micronaut.core.annotation.Introspected;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;

@Entity @Introspected
public class SomeEntity {

    @Column private Long id;
    @Id @Column private String transactid;

    public SomeEntity() { }
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getTransactid() { return transactid; }
    public void setTransactid(String transactid) { this.transactid = transactid; }
}
```
2. Create repository with findBy and existBy 
```
package bug;
import io.micronaut.data.annotation.Repository;
import io.micronaut.data.jpa.repository.JpaRepository;
import java.util.Optional;

@Repository
public interface BaseRepository<E extends SomeEntity> extends JpaRepository<E, String> {
    Optional<E> findByTransactid(String transactid);
    boolean existsByTransactid(String transactid);
}
```
3. Create test
```
package bug;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import org.junit.jupiter.api.Test;
import javax.inject.Inject;
import static org.junit.jupiter.api.Assertions.assertFalse;

@MicronautTest class BaseRepositoryTest {
    @Inject private BaseRepository<SomeEntity> baseRepository;
    @Test void verifyFindBy() { assertFalse(baseRepository.findByTransactid("dummy").isPresent()); }
    @Test void verifyExistsBy() { assertFalse(baseRepository.existsByTransactid("dummy")); }
}
```
4. Update Micronaut 2.4.1 -> 2.5.0

### Expected Behaviour

findBy and existBy work in 2.5.0 as they work in 2.4.1.

As inspected via `io.micronaut.data.hibernate.operations.HibernateJpaOperations#findOne(io.micronaut.data.model.runtime.PreparedQuery<T,R>)`

In Micronaut 2.4.1 produces correctly the where clause `WHERE (someEntity_.transactid = :p1)` using the referenced column, in the example 'transactid':
```
Exists by: SELECT someEntity_.transactid FROM bug.SomeEntity AS someEntity_ WHERE (someEntity_.transactid = :p1)
Find by: SELECT someEntity_ FROM bug.SomeEntity AS someEntity_ WHERE (someEntity_.transactid = :p1)
```

### Actual Behaviour

findBy and existBy in Micronaut 2.5.0 reference incorrectly in the where clause the 'id' column, `WHERE (someEntity_.id = :p1)`:
```
Exists by: SELECT someEntity_.transactid FROM au.com.bug.SomeEntity AS someEntity_ WHERE (someEntity_.id = :p1)
Find by: SELECT someEntity_ FROM bug.SomeEntity AS someEntity_ WHERE (someEntity_.id = :p1)
```

Also in this case since the transactid property is of type String an exception is thrown on executing `someentity0_.id=?` trying to fill in transactid as String on a Long property:
```
Hibernate: select someentity0_.transactid as col_0_0_ from some_entity someentity0_ where someentity0_.id=?

class java.lang.String cannot be cast to class java.lang.Long (java.lang.String and java.lang.Long are in module java.base of loader 'bootstrap')
java.lang.ClassCastException: class java.lang.String cannot be cast to class java.lang.Long (java.lang.String and java.lang.Long are in module java.base of loader 'bootstrap')
	at org.hibernate.type.descriptor.java.LongTypeDescriptor.unwrap(LongTypeDescriptor.java:19)
	at org.hibernate.type.descriptor.sql.BigIntTypeDescriptor$1.doBind(BigIntTypeDescriptor.java:46)
	at org.hibernate.type.descriptor.sql.BasicBinder.bind(BasicBinder.java:73)
	at org.hibernate.type.AbstractStandardBasicType.nullSafeSet(AbstractStandardBasicType.java:276)
	at org.hibernate.type.AbstractStandardBasicType.nullSafeSet(AbstractStandardBasicType.java:271)
	at org.hibernate.param.NamedParameterSpecification.bind(NamedParameterSpecification.java:53)
	at org.hibernate.loader.hql.QueryLoader.bindParameterValues(QueryLoader.java:648)
	at org.hibernate.loader.Loader.bindPreparedStatement(Loader.java:2132)
	at org.hibernate.loader.Loader.prepareQueryStatement(Loader.java:2109)
	at org.hibernate.loader.Loader.executeQueryStatement(Loader.java:2041)
	at org.hibernate.loader.Loader.executeQueryStatement(Loader.java:2019)
	at org.hibernate.loader.Loader.doQuery(Loader.java:948)
	at org.hibernate.loader.Loader.doQueryAndInitializeNonLazyCollections(Loader.java:349)
	at org.hibernate.loader.Loader.doList(Loader.java:2850)
	at org.hibernate.loader.Loader.doList(Loader.java:2832)
	at org.hibernate.loader.Loader.listIgnoreQueryCache(Loader.java:2664)
	at org.hibernate.loader.Loader.list(Loader.java:2659)
	at org.hibernate.loader.hql.QueryLoader.list(QueryLoader.java:506)
	at org.hibernate.hql.internal.ast.QueryTranslatorImpl.list(QueryTranslatorImpl.java:400)
	at org.hibernate.engine.query.spi.HQLQueryPlan.performList(HQLQueryPlan.java:219)
	at org.hibernate.internal.SessionImpl.list(SessionImpl.java:1414)
	at org.hibernate.query.internal.AbstractProducedQuery.doList(AbstractProducedQuery.java:1625)
	at org.hibernate.query.internal.AbstractProducedQuery.list(AbstractProducedQuery.java:1593)
	at io.micronaut.data.hibernate.operations.HibernateJpaOperations.lambda$findOne$1(HibernateJpaOperations.java:183)
	at io.micronaut.transaction.support.AbstractSynchronousTransactionManager.executeRead(AbstractSynchronousTransactionManager.java:155)
	at io.micronaut.data.hibernate.operations.HibernateJpaOperations.findOne(HibernateJpaOperations.java:144)
	at io.micronaut.data.hibernate.operations.HibernateJpaOperations.exists(HibernateJpaOperations.java:264)
	at io.micronaut.data.runtime.intercept.DefaultExistsByInterceptor.intercept(DefaultExistsByInterceptor.java:48)
	at io.micronaut.data.runtime.intercept.DefaultExistsByInterceptor.intercept(DefaultExistsByInterceptor.java:33)
	at io.micronaut.data.intercept.DataIntroductionAdvice.intercept(DataIntroductionAdvice.java:80)
	at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:96)
	at io.micronaut.validation.ValidatingInterceptor.intercept(ValidatingInterceptor.java:137)
	at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:96)
	at au.com.bug.BaseRepository$Intercepted.existsByTransactid(Unknown Source)
	at au.com.bug.BaseRepositoryTest.verifyExistsBy(BaseRepositoryTest.java:13)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:688)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	[SNAP]
```

I also noticed if I change the property name from `id` to something else, the where clause correctly uses `someEntity_.transactid = :p1` again.
```...
public class SomeEntity {

    @Id @Column private String transactid;
    @Column private Long id2;
...
```

Arguable with the knowledge on the above, this could be worked around.
But we have a use case where we have an `@EntityListeners` to generate the value on the id property from a sequence, while the transactid column is our primary id. 
Micronaut Data does not support using `@GeneratedValue` and `@SequenceGenerator` on fields which are not annotated with `@Id`. Also, non-PK columns cannot have a default value like "next value for my_sequence",
 
### Environment Information

- **Operating System**: Mac
- **Micronaut Version:** 2.4.1 and 2.5.0
- **JDK Version:** 11

**Comment by dstepanov** [comment]
> Micronaut Data does not support using @GeneratedValue and @SequenceGenerator on fields which are not annotated with @Id. Also, non-PK columns cannot have a default value like "next value for my_sequence",

Is there a compile error? I see you are using JPA so this limitation shouldn't be present.

**Comment by StefanSchmidtOz** [comment]
@dstepanov thanks for that, I only recently inherited the code, so the above comment on @GeneratedValue and @SequenceGenerator was just to give a bit more background on why columns were named in such a way. I left the related code out since the issue with the where clause was reproducible without all that.

I've included now a stacktrace in the description.
It appears to be really an edge case where a Column annotated property is named `id` but not being the primary id.
As stated above renaming `id` to something else, e.g. `seqId` the correct where clause is produced again.
And for now we will be applying this as a workaround.
```
    @Id @Column private String transactid;
    @Column private Long seqId;
```
